public/src/modules/translator.js

```diff
@@ -295,11 +295,13 @@
 	if (typeof define === 'function' && define.amd) {
 		define('translator', translator);
 
+		var _translator = translator;
+
 		// Expose a global `translator` object for backwards compatibility
 		window.translator = {
 			translate: function() {
 				console.warn('[translator] Global invocation of the translator is now deprecated, please `require` the module instead.');
-				translator.translate.apply(translator, arguments);
```

Comment:
@julianlam I now realise that I was wrong. This should work. For some reason I thought that the function would set precedence on the window.translator object instead of the local translator object. It was late, I was tired. I thought it would result in an infinite recursion.